### PR TITLE
fix windows stdout messy code

### LINKED ISSUE #2545: set output to console cause messy code on windows
### Environment

* **Platform**:  win10 

### Suggested Fix:
use `SetConsoleOutputCP`   https://docs.microsoft.com/en-us/windows/console/setconsoleoutputcp
command use 
`tesseract E:\test\single_ch_line.jpg stdout -l chi_sim `

### PR DESCRIPTION
**Pull request by zhuangzhuang**
fix #2545

### CODE REVIEW

**Comment by egorpugin** [comment]
Please, remove global and create this object explicitly in main.
I think simple code below is enough:
```
#ifdef _WIN32
SetConsoleOutputCP(CP_UTF8);
#endif
```

File: src/api/tesseractmain.cpp [hunk 1]
```diff
@@ -69,6 +69,31 @@ static void Win32WarningHandler(const char* module, const char* fmt,
...
+      SetConsoleOutputCP(oldCP_);
+    }
+  }
+
+ private:
+  bool changed_;
+  UINT oldCP_;
+};
+
+static AutoWin32ConsoleOutputCP autoWin32CosnoleOutputCP(CP_UTF8);
```
Comment (stweil) [inline src/api/tesseractmain.cpp]: Typo in name

**Comment by zhuangzhuang** [comment]
> I think simple code below is enough for main():
> 
> ```
> #ifdef _WIN32
> SetConsoleOutputCP(CP_UTF8);
> #endif
> ```

@egorpugin 
i think this must handle on exit.
just  like https://github.com/microsoft/vcpkg/blob/master/toolsrc/src/vcpkg.cpp#L322

**Comment by egorpugin** [comment]
Sorry, did not understand what you mean.

File: src/api/tesseractmain.cpp [hunk 2]
```diff
@@ -69,6 +69,31 @@ static void Win32WarningHandler(const char* module, const char* fmt,
 }
 
 #endif /* HAVE_TIFFIO_H */
+
+class AutoWin32ConsoleOutputCP {
+ public:
+  explicit AutoWin32ConsoleOutputCP(UINT codeCP) : changed_(false) {
+    auto oldCP = GetConsoleOutputCP();
```
Comment (egorpugin) [inline src/api/tesseractmain.cpp]: Maybe it's possible to remove this? Just always set and unset, even if the same CP.

**Comment by zdenop** [comment]
@zhuangzhuang: should not something like this help to solve #1252?

**Pull request merged**